course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -220,6 +281,64 @@
             // return the filtered list
             return filteredResults;
         };
+        $scope.getFormattedCourseInstance = function(ci) {
+            // This is a helper function that formats the CourseInstance metadata
+            // and is combination of existing logic in
+            // Searchcontroller.courseInstanceToTable and Searchcontroller cell
+            // render functions.
+            var courseInstance = {};
+            if (ci) {
+                courseInstance['title']= ci.title;
+                courseInstance['school'] = ci.course ?
+                        ci.course.school_id.toUpperCase() : '';
+                courseInstance['term'] = ci.term ? ci.term.display_name : '';
+                courseInstance['year'] = ci.term ? ci.term.academic_year : '';
+                courseInstance['course_instance_id'] = ci.course_instance_id;
+                courseInstance['registrar_code_display'] = ci.course ?
+                        (ci.course.registrar_code_display
+                        || ci.course.registrar_code).trim() : '';
+                if (ci.secondary_xlist_instances &&
+                    ci.secondary_xlist_instances.length > 0) {
+                        courseInstance['xlist_status'] = 'Primary';
+                } else if (ci.primary_xlist_instances &&
+                    ci.primary_xlist_instances.length > 0) {
+                        courseInstance['xlist_status'] = 'Secondary';
+                } else {
+                        courseInstance['xlist_status'] = 'N/A';
+                }
+                var sites = ci.sites || [];
+                var siteIds =[];
+                sites.forEach(function (site) {
+                    site.site_id = site.external_id;
+                    if (site.site_id.indexOf('http') === 0) {
+                        site.site_id = site.site_id.substr(site.site_id.lastIndexOf('/')+1);
+                    }
+                    siteIds.push(site.site_id);
+                });
+                courseInstance['sites']= siteIds.length>0 ? siteIds.join(', ') : 'N/A';
+            }
+            return courseInstance;
+        };
+        $scope.getMembersByUserId = function(memberList) {
+            /* generates a lookup table/dict/object to find a member's profile
+             by univ_id; used e.g. for checking whether the univ_id found for
+             each person in the search term lookup results is already in the
+             course or not
+             */
+            var membersByUserId = {};
+            memberList.forEach(function(member) {
+                var memberCopy = angular.copy(member);
+                // this is a temp fix to change the display text of the role
+                // "Teaching Fellow" to "TA" a more perm solution is being
+                // discussed, but will involve talking to the schools.
+                if (member.role && member.role.role_name == "Teaching Fellow") {
```

Comment:
Just a suggestion...if we want to work with a copy, maybe refer to the copy everywhere in this method, not the original, so we're sure there's no chance we're modifying the original by mistake?